virtcontainers/factory.go

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) 2018 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+type Factory interface {
```

Comment:
@bergwolf I agree with @amshinde here. The difference between `Factory` and `FactoryBase` is very thin and confusing, and unless there's a very good reason to keep it, you should simplify this with `Factory` interface only (basically what you've defined in `FactoryBase` right now).

Reply:
@sboeuf That would push the vm config comparison and CPU/memory hotplug down to each of the factory implementations. `factory/factory.go` is not just a broker but also handles vm config validation/comparison and possible CPU/memory hotplugs. It's better to do it at the factory level instead of doing the same work in each of the factory implementations.